chore: Homepage Rework (Issues #27 and #28)

penczak opened the pull request:
Fixed bug of background SVG waves zooming out and implemented design changes.
TODO: Change logo on homepage.

## Overview

Addresses Issues #27 and #28. 

I implemented the new desired design for the homepage and fixed the zoom bug. 

## What Changed

Go more into detail about key files that were modified and why they were
updated. Do not list every file that was modified; only note the ones most
relevant to this feature or bug fix.

pages/index.tsx: Changed the return to have only one card instead of two and to match the Figma.
components/common/waveSVG.tsx and waveSVG2.tsx: The zoom bug was due to something here. I just changed the width and height properties of the <svg> to be 100% rather than pixel values. 
Figma:
![image](https://user-images.githubusercontent.com/97853271/161429519-c54c328a-41cc-483b-8074-337174363cc0.png)

My version:
![image](https://user-images.githubusercontent.com/97853271/161429556-259947f5-10d8-4847-ad65-794dfda87bb5.png)

The only thing that still needs to be changed is the logo, but Tayisiya 

## Other Notes

In line 38 of index.tsx, I added a <br/> to get the spacing right on the logo, but this doesn't seem like a good practice. I couldn't get normal margins to work, so I did this as a work around.

eboysen commented on components/common/splashPageSearchBar.tsx:
```diff
@@ -133,8 +133,8 @@ export const SplashPageSearchBar = (props: SearchProps) => {
-                className="rounded-md border-primary-dark border-2 w-full h-12 pl-12 bg-white text-primary-dark placeholder-primary-dark font-bold"
-                placeholder="Search section number, professor name, course number...."
+                className="rounded-md border-primary-dark border-2 w-full h-12 pl-12 bg-white text-primary-dark placeholder-primary-light font-bold"
+                placeholder="CS1337..."
```
This probably does not need to be changed to match the figma exactly, previous placeholder was fine too

eboysen commented on pages/index.tsx:
```diff
@@ -30,27 +30,24 @@ const Home: NextPage = () => {
+          <div className="w-full relative m-auto">
+            <Card className="bg-light relative overflow-visible drop-shadow-lg rounded-xl bg-opacity-75 m-auto sm:w-5/12 xs:w-11/12 md:w-3/5 ">
+              <div className='' style={{margin: "1rem"}}>
+                <div className="m-auto w-1/6" style={{margin: "1rem"}}>
```
Try adding a color: "black" to the style tag on this div. The SVG should be set to be text color responsive, this will help prevent the bordered look of the logo.

eboysen reviewed (changes requested):
Regarding issue #27, try to get the logo to also scale with zooming, that seems to be the only component that is still not scaling properly. Also see other code comments regarding adjusting logo.

penczak commented:
Ok, I changed to logo color and changed the placeholder text back. I kept the "light" placeholder text because I think it looks much better, but if you want me to change that back to the dark, I can.

eboysen commented:
Looks good on my end, clean up the mobile experience a little bit too. I'm gonna assign Shoaib to take a look real quick too since he was the original person to submit the bug report to see if it matches what he expected.

shoaibhuq commented:
Regarding Issue #27 

At 100% the font size for the title is much larger than what I desired in the Figma.
![image](https://user-images.githubusercontent.com/44373355/161451361-d2e01ced-7f90-41a2-9853-1be707b12b41.png)

At 25% you can see clearly that the text label is not scaling currently
![image](https://user-images.githubusercontent.com/44373355/161451406-672c6bf2-8103-4e0c-9491-9304e1a173c3.png)



Regarding #28 

![image](https://user-images.githubusercontent.com/44373355/161451713-7b7cfb8d-131d-4c17-b0af-73077f718bc1.png)![image](https://user-images.githubusercontent.com/44373355/161451704-ec0be9a2-00cb-4dc3-9e59-6f198f8c75bf.png)

There are a few issues between what is desired in the design and the current deployment:

- Background gradient is incorrect currently, it should go from colors: `7486CE` to `BCC9FD`
- The white backcard should be larger
- Title Label 
>Use font size for `Heading 1` 
- Textview box should be larger
>Font size in text view box should be `Body 1`

Refrence for Font Size
![image](https://user-images.githubusercontent.com/44373355/161451939-67da7019-c88b-46c7-87e1-8264a0c7ad05.png)

shoaibhuq reviewed (changes requested):
https://github.com/UTDNebula/athena/pull/34#issuecomment-1086965458

eboysen reviewed (changes requested):
Almost there, found one more thing in need of fixing. For devices where the screen height is smaller than the grey box, the box overflows and the background is cut off. This should be fixable based upon using a percentage of height rather than a static one for the box so that it stays on a single screen. If you want to replicate, select a mobile device in the responsive layouts and rotate it to landscape
![image](https://user-images.githubusercontent.com/50427509/161863883-40e99304-b79c-425b-864e-41b17370d3b0.png)

penczak commented:
> Almost there, found one more thing in need of fixing. For devices where the screen height is smaller than the grey box, the box overflows and the background is cut off. 

Good catch.. I would not have noticed that. Should be fixed now.

penczak commented:
So to merge with the StyleGuidist branch, will I just close this PR and make another one after @eboysen  pushes the StyleGuidist directory changes?

eboysen commented:
> So to merge with the StyleGuidist branch, will I just close this PR and make another one after @eboysen pushes the StyleGuidist directory changes?

Yes, we likely will go about it like that, I will wait until a new branch and PR open up before closing anything out though
